tests/scripts/components-platform.sh

```diff
@@ -0,0 +1,121 @@
+# components-platform.sh
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+
+# This file contains the test components that are executed by all.sh
+
+################################################################
+#### Platform Testing
+################################################################
+
+component_test_malloc_0_null () {
```

Comment:
`component_test_malloc_0_null` is testing a simulated platform, like `component_test_no_platform`. It's unlike the platform-specific tests that make up the rest of the platform group. The two components belong in the same group, presumably the generic configuration group. Although maybe we should distinguish platform support features (`memory_buffer_alloc`, `malloc_0`, `no_platform`, `baremetal`, etc.) from miscellaneous configuration testing (`default`, `test_ref_configs`, …)? And as I mentioned in another comment, probably split platform into platform-x86 and platform-arm.